core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java

```diff
@@ -466,6 +466,39 @@ private void applyWalParallel(ObjList<FuzzTransaction> transactions, String tabl
         applyManyWalParallel(tablesTransactions, applyRnd, tableName, false);
     }
 
+    private void assertRandomIndexes(String tableNameNoWal, String tableNameWal, Rnd rnd) throws SqlException {
+        try (TableReader reader = newTableReader(configuration, tableNameNoWal)) {
+            if (reader.size() > 0) {
+                TableReaderMetadata metadata = reader.getMetadata();
+                int columnIndex = rnd.nextInt(metadata.getColumnCount());
+                for (int i = 0; i < metadata.getColumnCount(); i++) {
+                    columnIndex += i;
+                    columnIndex = columnIndex % metadata.getColumnCount();
+                    if (ColumnType.isSymbol(metadata.getColumnType(columnIndex))
+                            && metadata.isColumnIndexed(columnIndex)) {
+                        break;
+                    }
+
+                    if (i == metadata.getColumnCount() - 1) {
+                        // No indexed symbols
+                        return;
+                    }
+                }
+
+                String columnName = metadata.getColumnName(columnIndex);
+                long randomRow = rnd.nextLong(reader.size());
+                sink.clear();
+                TestUtils.printSql(compiler, sqlExecutionContext, "select \"" + columnName + "\" as a from " + tableNameNoWal + " limit " + randomRow + ", 1", sink);
+                String prefix = "a\n";
+                String randomValue = sink.length() > prefix.length() + 2 ? sink.subSequence(prefix.length(), sink.length() - 1).toString() : null;
+                String indexedWhereClause = " where \"" + columnName + "\" = " + (randomValue == null ? "null" : "'" + randomValue + "'");
+                LOG.info().$("checking random index with filter: ").$(indexedWhereClause).I$();
+                String limit = " limit 8970, 9100";
```

Comment:
nit: would be nice to add a comment on why we use exactly these limit boundaries here

Reply:
ah, that's debugging, removing

Reply:
removed at [a92f9ae](https://github.com/questdb/questdb/pull/3266/commits/a92f9ae47f5978081174d69a3831d8cca0cedf40)